twisted/python/_shellcomp.py

```diff
@@ -174,11 +174,12 @@ def write(self, genSubs=True):
             gen = ZshArgumentsGenerator(self.options, self.cmdName, self.file)
             gen.extraActions.insert(0, SubcommandAction())
             gen.write()
-            self.file.write('local _zsh_subcmds_array\n_zsh_subcmds_array=(\n')
+            self.file.write(b'local _zsh_subcmds_array\n_zsh_subcmds_array=(\n')
             for (cmd, short, parser, desc) in self.options.subCommands:
-                self.file.write('"%s:%s"\n' % (cmd, desc))
-            self.file.write(")\n\n")
-            self.file.write('_describe "sub-command" _zsh_subcmds_array\n')
+                self.file.write(
+                    b'"%s:%s"\n' % (cmd.encode('utf-8'), desc.encode('utf-8')))
```

Comment:
The build on Python 3.3 and Python 3.4 is failing here: https://travis-ci.org/twisted/twisted/jobs/144688549#L2298 .  You need to get rid of % formatting here also